services/base.spec.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+
+const BaseService = require('./base');
+
+class DummyService extends BaseService {
+  async handle({someArg}) {
+    return {
+      text: 'Hello ' + someArg,
+    };
+  }
+
+  static get category() { return 'cat'; }
+  static get uri() {
+    return {
+      format: '/foo/([^/]+)',
+      capture: ['someArg']
+    }
+  }
+}
+
+const expectedRouteRegex = /^\/foo\/([^\/]+).(svg|png|gif|jpg|json)$/;
+
+describe('BaseService', () => {
+  let mockCamp;
+  let mockHandleRequest;
+
+  beforeEach(() => {
+    mockCamp = {
+      route: sinon.spy(),
+    };
+    mockHandleRequest = sinon.spy();
+    DummyService.register(mockCamp, mockHandleRequest);
+  });
+
+  it('registers the service', () => {
+    assert(mockCamp.route.calledOnce);
+    assert.equal(mockCamp.route.getCall(0).args[0].toString(), expectedRouteRegex);
+  });
+
+  it('handles the request', async () => {
```

Comment:
Aw, love these async tests with async/await!

Reply:
Yeah I'm happy that Mocha allows test cases to return promises. I'm used to Jest, so I had to keep the Mocha docs open while writing these tests. They're similar but there's subtle differences (eg. Mocha doesn't have a mocking/stubbing framework built in and you need to use Sinon, whereas Jest includes mocking out-of-the-box).